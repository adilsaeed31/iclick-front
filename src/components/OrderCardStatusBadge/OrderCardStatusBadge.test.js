import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "custom/reactionTheme";
import OrderCardStatusBadge from "./OrderCardStatusBadge";

const statusCanceled = {
  label: "Canceled",
  status: "coreOrderWorkflow/canceled"
};

const statusNew = {
  label: "New",
  status: "new"
};

const statusProcessing = {
  label: "Processing",
  status: "coreOrderWorkflow/processing"
};

const statusCompleted = {
  label: "Completed",
  status: "coreOrderWorkflow/completed"
};

test("basic snapshot of canceled status", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <OrderCardStatusBadge
        status={statusCanceled}
      />
    </MuiThemeProvider>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot of new status", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <OrderCardStatusBadge
        status={statusNew}
      />
    </MuiThemeProvider>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot of processing status", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <OrderCardStatusBadge
        status={statusProcessing}
      />
    </MuiThemeProvider>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot of completed status", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <OrderCardStatusBadge
        status={statusCompleted}
      />
    </MuiThemeProvider>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
