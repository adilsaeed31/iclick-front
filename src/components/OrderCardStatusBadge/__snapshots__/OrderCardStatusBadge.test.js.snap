// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot of canceled status 1`] = `
<div
  className="MuiChip-root-5"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={-1}
>
  <span
    className="MuiChip-label-24"
  />
</div>
`;

exports[`basic snapshot of completed status 1`] = `
<div
  className="MuiChip-root-5"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={-1}
>
  <span
    className="MuiChip-label-24"
  />
</div>
`;

exports[`basic snapshot of new status 1`] = `
<div
  className="MuiChip-root-5"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={-1}
>
  <span
    className="MuiChip-label-24"
  />
</div>
`;

exports[`basic snapshot of processing status 1`] = `
<div
  className="MuiChip-root-5"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={-1}
>
  <span
    className="MuiChip-label-24"
  />
</div>
`;
